src/libstd/sys/unix/ext/net.rs

```diff
@@ -90,6 +92,35 @@ unsafe fn sockaddr_un(path: &Path) -> io::Result<(libc::sockaddr_un, libc::sockl
     Ok((addr, len as libc::socklen_t))
 }
 
+unsafe fn get_fds_from_msghdr(msg: &libc::msghdr, fds: &mut [RawFd]) -> usize {
+    let fds_len = fds.len();
+    let mut count_fd = 0;
+
+    let cmsg_len_zero = libc::CMSG_LEN(0) as usize;
+    let mut cmsg = libc::CMSG_FIRSTHDR(msg);
+
+    while !cmsg.is_null() && count_fd < fds_len {
+        if (*cmsg).cmsg_level == libc::SOL_SOCKET && (*cmsg).cmsg_type == libc::SCM_RIGHTS {
+            let data_len = (*cmsg).cmsg_len - cmsg_len_zero;
+            let data = libc::CMSG_DATA(cmsg) as *const RawFd;
```

Comment:
I like to use pointer::cast as it maintains the mutability.

Reply:
What do you mean with pointer::cast? Can you show me an example, please?

Reply:
https://doc.rust-lang.org/stable/std/primitive.pointer.html#method.cast-1